edge-hub/test/Microsoft.Azure.Devices.Routing.Core.Test/RouteTest.cs

```diff
@@ -19,29 +20,29 @@ public class RouteTest : RoutingUnitTestBase
-        static readonly Route Route5 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>());
-        static readonly Route Route6 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>());
-        static readonly Route Route7 = new Route("id2", "rule1", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { Endpoint1 });
+        static readonly Route Route1 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
+        static readonly Route Route2 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint2, 0, 0);
+        static readonly Route Route3 = new Route("id1", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint3, 0, 0);
+        static readonly Route Route4 = new Route("id2", "rule2", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
+        static readonly Route Route5 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0);
+        static readonly Route Route6 = new Route("id3", "rule3", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0);
+        static readonly Route Route7 = new Route("id2", "rule1", "hub", TelemetryMessageSource.Instance, Endpoint1, 0, 0);
 
         [Fact]
         [Unit]
         public void TestConstructor()
         {
-            Assert.Throws<ArgumentNullException>(() => new Route(null, "condition", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>()));
-            Assert.Throws<ArgumentNullException>(() => new Route("id", null, "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint>()));
-            Assert.Throws<ArgumentNullException>(() => new Route("id", "condition", null, TelemetryMessageSource.Instance, new HashSet<Endpoint>()));
-            Assert.Throws<ArgumentNullException>(() => new Route("id", "condition", "hub", TelemetryMessageSource.Instance, null));
+            Assert.Throws<ArgumentNullException>(() => new Route(null, "condition", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0));
+            Assert.Throws<ArgumentNullException>(() => new Route("id", null, "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0));
+            Assert.Throws<ArgumentNullException>(() => new Route("id", "condition", null, TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0));
+            Assert.Throws<ArgumentNullException>(() => new Route("id", "condition", "hub", TelemetryMessageSource.Instance, null, 0, 0));
         }
 
         [Fact]
         [Unit]
         public void SmokeTest()
         {
-            var route = new Route("id", "true", "hub", TelemetryMessageSource.Instance, new HashSet<Endpoint> { new TestEndpoint("id1") });
+            var route = new Route("id", "true", "hub", TelemetryMessageSource.Instance, new Mock<Endpoint>("id1").Object, 0, 0);
```

Comment:
nit:It seems all test cases in this class, use zero for both TTL and priority.  Please just use different values.

Reply:
that's because the tests don't verify TTL and priority, so they're just fill in values much like mock endpoints.  Tests that actually cover TTL and priority will have more interesting values, and that'll come in a separate WI.

---
In reply to: [377462568](https://github.com/Azure/iotedge/pull/2469#discussion_r377462568) [](ancestors = 377462568)